### PR TITLE
[ARTS-531] - Discovery server

### PR DESCRIPTION
**Pull request by eladiw**
## Description
Integrating the Discovery Server + wiring the different micro-services.

### Changes
- [ARTS-531] - Deploying the discovery server and configuration changes to wire it

### Checklist:

- [x] Branch name follows convention (feature/arts-###-short-name OR fix/arts-###-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

[ARTS-531]: https://ndph-arts.atlassian.net/browse/ARTS-531

### CODE REVIEW

File: terraform/trial_rg/services.tf [hunk 1]
```diff
@@ -12,18 +22,29 @@ module "trial_app_service_site" {
   docker_image_tag    = var.site_image_tag
 
   settings = {
-    "WEBSITES_PORT"               = "8080" # The container is listening on 8080
+    "SPRING_APPLICATION_NAME"                 = "site-service"
```
Comment (tamirkamara) [inline terraform/trial_rg/services.tf]: You're overriding the default application name that's packaged in the source. Is it required?

File: terraform/trial_rg/services.tf [hunk 2]
```diff
@@ -12,18 +22,29 @@ module "trial_app_service_site" {
...
+    "SPRING_PROFILES_ACTIVE"                  = var.spring_profile
+    "SPRING_CLOUD_CONFIG_LABEL"               = var.spring_config_label
+    "SERVER_PORT"                             = "80"
+    "WEBSITES_PORT"                           = "80"
+    "SPRING_CLOUD_CONFIG_DISCOVERY_ENABLED"   = "true"
+    "SPRING_CLOUD_CONFIG_DISCOVERY_SERVICEID" = "config-server"
```
Comment (tamirkamara) [inline terraform/trial_rg/services.tf]: This is making the profiles redundant... should we do this?

File: terraform/trial_rg/services.tf [hunk 3]
```diff
@@ -12,18 +22,29 @@ module "trial_app_service_site" {
...
+    "EUREKA_CLIENT_SERVICEURL_DEFAULTZONE"    = "https://${module.trial_sc_discovery.name}.azurewebsites.net/eureka/"
```
Comment (tamirkamara) [inline terraform/trial_rg/services.tf]: Can we use a direct output of the terraform module to get this URL, maybe default_site_hostname?

**Review by tamirkamara — CHANGES_REQUESTED** [review]
My comments are relevant for all websites but I only wrote them once

File: terraform/trial_rg/services.tf (hunk 1 above)
Comment (eladiw) [reply terraform/trial_rg/services.tf]: no

File: terraform/trial_rg/services.tf (hunk 2 above)
Comment (eladiw) [reply terraform/trial_rg/services.tf]: correct

File: terraform/trial_rg/services.tf (hunk 3 above)
Comment (eladiw) [reply terraform/trial_rg/services.tf]: yes

File: terraform/trial_rg/services.tf [hunk 4]
```diff
@@ -33,23 +50,25 @@ module "trial_app_service_practitioner" {
...
   settings = {
-    "SPRING_APPLICATION_NAME"     = "practitioner-service"
-    "SPRING_PROFILES_ACTIVE"      = var.spring_profile
-    "SPRING_CLOUD_CONFIG_LABEL"   = var.spring_config_label
-    "SPRING_CLOUD_CONFIG_URI"     = "https://${module.trial_sc_config.name}.azurewebsites.net"
-    "WEBSITES_PORT"               = "8080" # The container is listening on 8080
+    "SPRING_PROFILES_ACTIVE"                  = var.spring_profile
+    "SPRING_CLOUD_CONFIG_LABEL"               = var.spring_config_label
+    "SERVER_PORT"                             = "80"
+    "WEBSITES_PORT"                           = "80" # The container is listening on 8080
```
Comment (eladiw) [inline terraform/trial_rg/services.tf]:
```suggestion
    "WEBSITES_PORT"                           = "80"
```

File: terraform/trial_rg/services.tf [hunk 5]
```diff
@@ -62,20 +81,27 @@ module "trial_app_service_role" {
...
     azurerm_app_service_plan.apps_service_plan,
     module.roles_sql_server,
+    module.trial_sc_config,
+    module.trial_sc_discovery,
   ]
 }
 
 # init service
 # todo: make 1-time service: ARTS-362
 module "trial_app_service_init" {
```
Comment (anatbal) [inline terraform/trial_rg/services.tf:102]: In line 117 (Unable to comment there) why on the depends_on there is no  module.trial_sc_config, module.trial_sc_discovery? The init is connected to the discovery...

**Review by anatbal — COMMENTED** [review]
Looks good
minor comment

File: terraform/trial_rg/services.tf (hunk 5 above)
Comment (eladiw) [reply terraform/trial_rg/services.tf:102]: will add. the init is in a WIP status so I didn't change it...

**Review by anatbal — DISMISSED** [review]

**Review by tamirkamara — DISMISSED** [review]

**Review by tamirkamara — APPROVED** [review]

**Review by anatbal — APPROVED** [review]

**Pull request merged**